de.metas.business/src/main/java/org/adempiere/pricing/api/impl/CalculateDiscountCommand.java

```diff
@@ -96,6 +105,92 @@ else if (X_M_DiscountSchema.DISCOUNTTYPE_Formula.equals(discountType)
 
 	}
 
+	private DiscountResult computeIndividualPriceForDiscountSchemaBreak(final I_M_DiscountSchemaBreak breakApplied)
+	{
+		final String priceBase = breakApplied.getPriceBase();
+
+		if (X_M_DiscountSchemaBreak.PRICEBASE_PricingSystem.equals(priceBase))
+		{
+
+			final I_M_ProductPrice productPrice = findProductPrice(breakApplied);
+
+			final BigDecimal priceStd = productPrice.getPriceStd();
+			final BigDecimal priceList = productPrice.getPriceList();
+			final BigDecimal priceLimit = productPrice.getPriceLimit();
+
+			final BigDecimal stdAddAmt = breakApplied.getStd_AddAmt();
+
+			return DiscountResult.builder()
+					.priceListOverride(priceList)
+					.priceLimitOverride(priceLimit)
+					.priceStdOverride(priceStd.add(stdAddAmt))
+					.build();
+
+		}
+		else if (X_M_DiscountSchemaBreak.PRICEBASE_Fixed.equals(priceBase))
+		{
+			final BigDecimal discountSchemaBreakStdPrice = breakApplied.getPriceStd();
+
+			return DiscountResult.builder()
+					.priceStdOverride(discountSchemaBreakStdPrice)
+					.build();
+		}
+		else
+		{
+			throw new AdempiereException("PriceBase value unknown: " + priceBase);
+		}
+	}
+
+	private I_M_ProductPrice findProductPrice(final I_M_DiscountSchemaBreak breakApplied)
```

Comment:
Hmm, I was thinking that we shall also support all other pricing rules like HU pricing, attribute pricing etc.
For that reason, i would suggest to:
* introduce `pricingContext' into CalculateDiscountRequest
* when fetching the prices from base pricing system, try calling `IPricingBL.calculatePrice(IPricingContext)` with a context which was created based on CalculateDiscountRequest's pricingContext, but with the correct pricing system.